src/nav/nav_model/nav_model_rings.py

```
@@ -33,3 +48,875 @@ def __init__(self,
+            #         self._logger.warning(
+            #             f'Feature {feature_key} {edge_type}_data[{i}] '
+            #             f'mode 1 missing {field}, skipping')
+            #         return False
+            # Validate a is positive
+            if 'a' in mode:
+                if not min_radius <= mode['a'] <= max_radius:
+                    return False
+                if mode['a'] <= 0:
+                    self._logger.warning(
+                        f'Feature {feature_key} {edge_type}_data[{i}] mode 1 '
+                        f'has non-positive a={mode["a"]}, skipping')
```

Ruff (0.14.11)

Logging statement uses f-string

(G004)